tests/unit/test_link_metrics_manager.cpp

```diff
@@ -0,0 +1,217 @@
+                                          int8_t               aRssiValue)
+{
+    aLinkMetricsValues.mMetrics.mLinkMargin = true;
+    aLinkMetricsValues.mMetrics.mRssi       = true;
+
+    aLinkMetricsValues.mLinkMarginValue = aLinkMarginValue;
+    aLinkMetricsValues.mRssiValue       = aRssiValue;
+}
+
+void UnitTester::TestLinkMetricsManager(void)
+{
+    ChildTable                  *childTable;
+    const uint16_t               testListLength = GetArrayLength(mTestChildList);
+    Ip6::Address                 linkLocalAddr;
+    LinkMetricsManager::Subject *subject1 = nullptr;
+    LinkMetricsManager::Subject *subject2 = nullptr;
+
+    sInstance = testInitInstance();
+    VerifyOrQuit(sInstance != nullptr);
+
+    childTable = &sInstance->Get<ChildTable>();
+
+    sInstance->Get<Mle::Mle>().SetRole(Mle::kRoleRouter);
+    // Add the child entries from test list
+    for (uint16_t i = 0; i < testListLength; i++)
+    {
+        Child *child;
+
+        child = childTable->GetNewChild();
+        VerifyOrQuit(child != nullptr, "GetNewChild() failed");
+
+        child->SetState(mTestChildList[i].mState);
+        child->SetExtAddress((AsCoreType(&mTestChildList[i].mExtAddress)));
+        child->SetVersion(mTestChildList[i].mThreadVersion);
+    }
+
+    LinkMetricsManager &linkMetricsMgr = sInstance->Get<LinkMetricsManager>();
+    linkMetricsMgr.SetEnabled(true);
+    // Update the subjects for the first time.
+    linkMetricsMgr.UpdateSubjects();
+
+    // Expect there are 2 subjects in NotConfigured state.
+    uint16_t index = 0;
+    for (LinkMetricsManager::Subject &subject : linkMetricsMgr.mSubjectList)
+    {
+        VerifyOrQuit(subject.mExtAddress == AsCoreType(&mTestChildList[1 - index].mExtAddress));
+        VerifyOrQuit(subject.mState == LinkMetricsManager::SubjectState::kNotConfigured);
+        index++;
+    }
+    subject1 = linkMetricsMgr.mSubjectList.FindMatching(AsCoreType(&mTestChildList[0].mExtAddress));
+    subject2 = linkMetricsMgr.mSubjectList.FindMatching(AsCoreType(&mTestChildList[1].mExtAddress));
+    VerifyOrQuit(subject1 != nullptr);
+    VerifyOrQuit(subject2 != nullptr);
+
+    // Update the state of the subjects
+    linkMetricsMgr.UpdateLinkMetricsStates();
+    // Expect the 2 subjects are in Configuring state.
+    for (LinkMetricsManager::Subject &subject : linkMetricsMgr.mSubjectList)
+    {
+        VerifyOrQuit(subject.mState == LinkMetricsManager::SubjectState::kConfiguring);
+    }
+
+    // subject1 received a response with a success status code
+    linkLocalAddr.SetToLinkLocalAddress(AsCoreType(&mTestChildList[0].mExtAddress));
+    linkMetricsMgr.HandleMgmtResponse(&linkLocalAddr, MapEnum(LinkMetrics::Status::kStatusSuccess));
+    VerifyOrQuit(subject1->mState == LinkMetricsManager::SubjectState::kActive);
+
+    // subject1 received Enhanced ACK IE and updated the link metrics data
+    otLinkMetricsValues linkMetricsValues;
```

Comment:
Add `{}` block to create scope for new variable definitions?

Reply:
This variable is almost used until the end. Is it better to put it at the begining or add `{}` for all the code next?